src/client/scripts/esm/chess/logic/movepiece.js

```diff
@@ -452,123 +308,84 @@ function rewindGameToIndex(gamefile, moveIndex, { removeMove = true, updateData
- * @param {Object} options - An object that may contain the properties:
- * - `doGameOverChecks`: Whether, while simulating this move, to perform game over checks such as checkmate or other win conditions. SLOWER, but this can be used to verify the game conclusion the opponent claimed. Default: *false*
- * @returns {Object} An object that may contains the properties:
- * - `isCheck`: Whether making this move puts the specified color in check. Usually it's you stepping into check.
- * - `gameConclusion`: The resulting `gameConclusion` after the move, if `doGameOverChecks` was specified as *true*.
+ * Wraps a function in a simulated move
+ * @param {gamefile} gamefile 
+ * @param {Move} move 
+ * @param {CallableFunction} callback
+ * @returns whatever is returned by the callback
  */
-function simulateMove(gamefile, move, colorToTestInCheck, { doGameOverChecks = false } = {}) {
+function simulateMoveWrapper(gamefile, move, callback) {
 	// Moves the piece without unselecting it or regenerating the pieces model.
-	makeMove(gamefile, move, { pushClock: false, animate: false, updateData: false, simulated: true, doGameOverChecks, updateProperties: doGameOverChecks });
-    
+	generateMove(gamefile, move);
+	makeMove(gamefile, move);
+
 	// What info can we pull from the game after simulating this move?
-	const info = {
-		isCheck: doGameOverChecks ? gamefile.inCheck : checkdetection.detectCheck(gamefile, colorToTestInCheck, []),
-		gameConclusion: doGameOverChecks ? gamefile.gameConclusion : undefined
-	};
+	const info = callback();
 
 	// Undo the move, REWIND.
 	// We don't have to worry about the index changing, it is the same.
 	// BUT THE CAPTURED PIECE MUST be inserted in the exact location!
 	// Only remove the move
-	rewindMove(gamefile, { updateData: false, animate: false });
+	rewindMove(gamefile, true);
+
+	return info;
+}
 
-	return info; // Info from simulating the move: { isCheck, gameConclusion }
+/**
+ * Simulates a move to get the check
+ * @param {gamefile} gamefile 
+ * @param {Move} move 
+ * @param {*} colorToTestInCheck 
+ * @returns 
+ */
+function getSimulatedCheck(gamefile, move, colorToTestInCheck) {
+	return simulateMoveWrapper(
+		gamefile,
+		move,
+		() => checkdetection.detectCheck(gamefile, colorToTestInCheck, []),
+	);	
 }
 
 /**
- * Strips the coordinates of their special move properties.
- * For example, unstripped coords may look like: `[2,7,enpassant:true]`
- * @param {number[]} coords - The coordinates
- * @returns {number[]} The stripped coordinates: `[2,7]`
+ * Simulates a move to get the gameConclusion
+ * @param {gamefile} gamefile 
+ * @param {Move} move 
+ * @returns the gameConclusion
  */
-function stripSpecialMoveTagsFromCoords(coords) { return [coords[0], coords[1]]; }
+function getSimulatedConclusion(gamefile, move) {
+	return simulateMoveWrapper(
+		gamefile,
+		move,
+		() => wincondition.getGameConclusion(gamefile)
```

Comment:
If makeMove() does game over checks by default, isn't this a repeated call to do the game over checks? Could we just check the gameConclusion on the gamefile?